packages/@lwc/template-compiler/src/codegen/codegen.ts

```diff
@@ -482,6 +489,10 @@ export default class CodeGen {
         }
 
         const scope = this;
+
+        // Cloning here is necessary because `this.replace()` is destructive, and we might use the
+        // node later during static content optimization
+        expression = doStructuredClone(expression);
```

Comment:
This is a really subtle bug – the code below is mutating the AST, which only becomes a problem later if we try to re-use it, which we do during the static content optimization, which is a second pass on top of the first pass we already did.

First pass:

https://github.com/salesforce/lwc/blob/650305a7f0a43d64c7286a39aeb9cd8891e9f283/packages/%40lwc/template-compiler/src/codegen/index.ts#L642

Second pass:

https://github.com/salesforce/lwc/blob/650305a7f0a43d64c7286a39aeb9cd8891e9f283/packages/%40lwc/template-compiler/src/codegen/codegen.ts#L572